internal/pkg/warewulfd/api/container.go

```diff
@@ -0,0 +1,197 @@
+	Writable bool     `json:"writable"`
+}
+
+func NewContainer(name string) *Container {
+	c := new(Container)
+	c.Kernels = []string{}
+	for _, k := range kernel.FindKernels(name) {
+		c.Kernels = append(c.Kernels, k.Path)
+	}
+	c.Size = container.ImageSize(name)
+	c.Writable = container.IsWriteAble(name)
+	return c
+}
+
+func getContainers() usecase.Interactor {
+	u := usecase.NewInteractor(func(ctx context.Context, _ struct{}, output *map[string]*Container) error {
+		m := make(map[string]*Container)
+		if names, err := container.ListSources(); err != nil {
+			return err
+		} else {
+			for _, name := range names {
+				m[name] = NewContainer(name)
+			}
+			*output = m
+			return nil
+		}
+	})
+	u.SetTitle("Get container images")
+	u.SetDescription("Get all container images.")
+	u.SetTags("Container")
+	return u
+}
+
+func getContainerByName() usecase.Interactor {
+	type getContainerByNameInput struct {
+		Name string `path:"name"`
+	}
+
+	u := usecase.NewInteractor(func(ctx context.Context, input getContainerByNameInput, output *Container) error {
+		if !container.ValidSource(input.Name) {
+			return fmt.Errorf("container not found: %v", input.Name)
+		} else {
+			*output = *NewContainer(input.Name)
+			return nil
+		}
+	})
+	u.SetTitle("Get a container")
+	u.SetDescription("Get a container by its name.")
+	u.SetTags("Container")
+	return u
+}
+
+func importContainer() usecase.Interactor {
+	type importContainerInput struct {
+		Name        string `path:"name"`
+		URI         string `json:"uri" required:"true" description:"Docker registry URI to download container images"`
+		NoHttps     bool   `json:"nohttps" default:"false" description:"Whether to use https for the registry URI, default:'false'"`
+		OciUserName string `json:"ociuser" description:"Username for the registry URI, if needed"`
+		OciPassword string `json:"ocipassword" description:"Password for the registry URI, if needed"`
+		Build       bool   `json:"build" default:"true" description:"Whether to build the container image, default:'true'"`
+		SycUser     bool   `json:"syncuser" default:"true" description:"Whether to syncuser, default:'true'"`
+		Default     bool   `json:"default" default:"false" description:"Whether to make the contaienr default, default:'false'"`
+		Force       bool   `json:"force" default:"false" description:"Whether to overwrite the existing container with the same name, default:'false'"`
+	}
+
+	u := usecase.NewInteractor(func(ctx context.Context, input importContainerInput, output *Container) error {
+		if !strings.HasPrefix(input.URI, "docker://") {
+			return errors.New("uri only supports docker:// prefix for now")
+		}
+
+		cip := &wwapiv1.ContainerImportParameter{
+			Source:      input.URI,
+			Name:        input.Name,
+			Force:       input.Force,
+			Build:       input.Build,
+			Default:     input.Default,
+			OciNoHttps:  input.NoHttps,
+			OciUsername: input.OciUserName,
+			OciPassword: input.OciPassword,
+			SyncUser:    input.SycUser,
+		}
```

Comment:
Please avoid using the grpc API code inside of this new API code, since the goal is to remove the grpc code. Refactor this (and potentially `container_api.ContainerImport`) to call a (potentially new) function in `internal/pkg/container/`. (Since importing from a directory doesn't really make sense from the web UI, it might be enough to just call `ImportDocker` directly.)

Reply:
oh, I saw that we have another issue FUZZ-3148 removing old gprc calls, so I am using the old way to unify the calling method. besides, it might be better to call the same function from both CLI and rest api? `ImportDocker` does not validate the input and we need additional similar validation code existed in `CotnainerImport`, also do we need to `Build/SyncUser` after `ImportDocker` is called?

Reply:
> it might be better to call the same function from both CLI and rest api

Definitely. I have something in mind, but I'll try my idea in the morning atop yours here in the morning. Please don't let this be a blocker for you, though; keep working on other API functions if there are any left on the list.